src/bhv_basic_move.cpp

```diff
@@ -105,13 +107,21 @@ Bhv_BasicMove::execute( PlayerAgent * agent )
     agent->debugClient().setTarget( target_point );
     agent->debugClient().addCircle( target_point, dist_thr );
 
-    if ( ! Body_GoToPoint( target_point, dist_thr, dash_power
-                           ).execute( agent ) )
-    {
-        Body_TurnToBall().execute( agent );
+
+    if ( Body_GoToPoint( target_point, dist_thr, dash_power
+			 ).execute( agent ) ||
+	 Body_TurnToBall().execute( agent ) ) {
+      if (ball.rposValid() || wm.self().collidesWithPost()) {
+	success = true;
+      } else {
+	success = false;
+      }
+    } else {
+      success = false;
     }
 
-    if ( wm.existKickableOpponent()
+    if ( wm.existKickableOpponent() &&
+	 ball.rposValid()
```

Comment:
I think the names of the two actions confused me; good point.. Actually, it looks like Neck_TurnToBall is *preferable* if ! ball.posValid(), since Neck_TurnToBallOrScan can otherwise actually wind up getting an invalid ball_next from agent->effector().queuedNextBallPos(), which it does not appear to be checking for - something that may need a bugfix (by default, Neck_TurnToBallOrScan checks the observation count vs 0...) Correcting.